File: apps/Skin-Image-Classification/Android/app/src/main/java/ai/zetic/skinclassifier/ui/Theme.kt
package ai.zetic.skinclassifier.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Dark, on-device-AI palette ported from the iOS app's `Theme.swift`: near-black background,
 * cyan→periwinkle brand gradient, and severity tints (mint / amber / coral).
 */
object Theme {
    val Bg = Color(0.04f, 0.05f, 0.08f)

    val Accent = Color(0.36f, 0.92f, 0.93f)   // cyan glow
    val Accent2 = Color(0.55f, 0.62f, 1.00f)  // periwinkle

    val Mint = Color(0.40f, 0.92f, 0.70f)
    val Amber = Color(1.00f, 0.78f, 0.36f)
    val Coral = Color(1.00f, 0.45f, 0.48f)

    val Ink = Color.White
    val InkSoft = Color.White.copy(alpha = 0.62f)
    val InkFaint = Color.White.copy(alpha = 0.40f)

    val Card = Color.White.copy(alpha = 0.05f)
    val Hairline = Color.White.copy(alpha = 0.12f)

    val Corner: Dp = 22.dp

    val brandGradient = Brush.horizontalGradient(listOf(Accent, Accent2))
}

/** Standard frosted "glass" card container. */
fun Modifier.glassCard(padding: Dp = 18.dp, corner: Dp = Theme.Corner): Modifier = this
    .fillMaxWidth()
    .clip(RoundedCornerShape(corner))
    .background(Theme.Card)
    .border(1.dp, Theme.Hairline, RoundedCornerShape(corner))
    .padding(padding)

File: apps/Skin-Image-Classification/iOS/SkinImageClassification/App/Theme.swift
import SwiftUI

/// Futuristic / glassy visual language for the Skin Image Classification demo.
///
/// Dark, near-black base with a deep teal→indigo aurora, frosted glass cards
/// (`.ultraThinMaterial`), a cyan medical-tech accent, and thin hairline strokes.
/// Centralized so every surface reads as one polished, professional system.
enum Theme {

    // MARK: Palette
    static let accent  = Color(red: 0.36, green: 0.92, blue: 0.93)   // cyan glow
    static let accent2 = Color(red: 0.55, green: 0.62, blue: 1.00)   // periwinkle
    static let mint    = Color(red: 0.40, green: 0.92, blue: 0.70)
    static let amber   = Color(red: 1.00, green: 0.78, blue: 0.36)
    static let coral   = Color(red: 1.00, green: 0.45, blue: 0.48)

    static let ink     = Color.white
    static let inkSoft = Color.white.opacity(0.62)
    static let inkFaint = Color.white.opacity(0.40)

    // MARK: Backgrounds

    /// Full-screen aurora gradient backdrop. Place behind everything.
    static var background: some View {
        ZStack {
            Color(red: 0.04, green: 0.05, blue: 0.08).ignoresSafeArea()
            // Soft aurora blobs
            RadialGradient(
                colors: [accent.opacity(0.22), .clear],
                center: .topLeading, startRadius: 8, endRadius: 460
            ).ignoresSafeArea()
            RadialGradient(
                colors: [accent2.opacity(0.20), .clear],
                center: .bottomTrailing, startRadius: 8, endRadius: 520
            ).ignoresSafeArea()
        }
    }

    /// Hairline stroke used on glass edges for that crisp glassmorphism rim.
    static var hairline: LinearGradient {
        LinearGradient(
            colors: [.white.opacity(0.45), .white.opacity(0.06)],
            startPoint: .topLeading, endPoint: .bottomTrailing
        )
    }

    /// Brand gradient for primary text / accents.
    static var brandGradient: LinearGradient {
        LinearGradient(colors: [accent, accent2], startPoint: .leading, endPoint: .trailing)
    }

    // MARK: Type
    static func mono(_ size: CGFloat, _ weight: Font.Weight = .medium) -> Font {
        .system(size: size, weight: weight, design: .monospaced)
    }
}

// MARK: - Glass card modifier

extension View {
    /// Frosted glass panel with a hairline rim and soft depth shadow.
    func glassCard(cornerRadius: CGFloat = 24, padding: CGFloat = 18) -> some View {
        self
            .padding(padding)
            .background(.ultraThinMaterial, in: RoundedRectangle(cornerRadius: cornerRadius, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)
                    .strokeBorder(Theme.hairline, lineWidth: 1)
            )
            .shadow(color: .black.opacity(0.35), radius: 18, x: 0, y: 10)
    }
}
